### PR TITLE
[WEJBHTTP-152] Refactor EJB server side

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WEJBHTTP-152

### CODE REVIEW

File: ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java [hunk 1]
```diff
@@ -0,0 +1,756 @@
...
+                        throw new RuntimeException("nyi");
+                    }
+                });
+                if(handle != null && identifier != null) {
+                    cancellationFlags.put(identifier, handle);
+                }
+            });
+        }
+
+        private static String handleDash(String s) {
```
Comment (rachmatowicz) [inline ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java:358]: Move handleDash to RemoteHTTPHandler as a common method.

File: ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java [hunk 2]
```diff
@@ -0,0 +1,756 @@
...
+                EjbHttpClientMessages.MESSAGES.debugf("Exchange %s did not include both session id and invocation id in cancel request", exchange);
+                return;
+            }
+            CancelHandle handle = cancellationFlags.remove(identifier);
+            if (handle != null) {
+                handle.cancel(cancelIdRunning);
+            }
+        }
+
+        private static String handleDash(String s) {
```
Comment (rachmatowicz) [inline ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java:524]: Move handleDash to RemoteHTTPHandler as a common method.

File: ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java [hunk 3]
```diff
@@ -0,0 +1,756 @@
...
+
+                    @Override
+                    public <C> C getProviderInterface(Class<C> providerInterfaceType) {
+                        return null;
+                    }
+                });
+            });
+        }
+
+        private static String handleDash(String s) {
```
Comment (rachmatowicz) [inline ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java:705]: Move handleDash to RemoteHTTPHandler as a common method.

**Review by rachmatowicz — COMMENTED** [review]
Made some comments. Overall, looks good to me.

File: ejb/src/main/java/org/wildfly/httpclient/ejb/EjbHttpService.java [hunk 4]
```diff
@@ -42,34 +42,22 @@
  *
  * @author Stuart Douglas
  * @author Flavia Rainone
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
  */
 public class EjbHttpService {
```
[the remark this answers is not in the document]
Comment (ropalka) [reply ejb/src/main/java/org/wildfly/httpclient/ejb/EjbHttpService.java]: Fixed.

File: ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java [hunk 5]
```diff
@@ -0,0 +1,756 @@
...
+            case CREATE_SESSION:
+                return new HttpSessionOpenHandler(association, executorService, ctx, config);
+            case DISCOVER:
+                return new HttpDiscoveryHandler(executorService, association, config);
+            default:
+                throw new IllegalStateException();
+        }
+    }
+
+    static final class HttpInvocationHandler extends RemoteHTTPHandler {
```
[the remark this answers is not in the document]
Comment (ropalka) [reply ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java]: Fixed.

File: ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java (hunk 2 above)
Comment (ropalka) [reply ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java]: Done.

File: ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java (hunk 1 above)
Comment (ropalka) [reply ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java]: Done.

File: ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java (hunk 3 above)
Comment (ropalka) [reply ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java]: Done.

**Comment by ropalka** [comment]
Thanks for valuable feedback @rachmatowicz ! Could you please review it again?

**Review by fl4via — APPROVED** [review]

**Pull request merged**